ProfilesApi/Services/Implementations/ReceptionistProfilesService.cs

```diff
@@ -0,0 +1,159 @@
+        }
+        var userIdStream = await createdUser.Content.ReadAsStreamAsync();
+        var userId = JsonSerializer.Deserialize<Guid>(userIdStream);
+        
+        var account = _mapper.Map<Account>(request);
+        account.UserId = userId;
+        //add createdBy UpdatedBy
+        try
+        {
+            await _accountRepository.CreateAsync(account);
+        }
+        catch
+        {
+            await _doctorService.RollBackUserAsync(account.UserId);
+            throw;
+        }
+        
+        var receptionist = _mapper.Map<Receptionist>(request);
+        receptionist.AccountId = account.Id;
+        try
+        {
+            await _receptionistRepository.CreateAsync(receptionist);
+        }
+        catch
+        {
+            await _doctorService.RollBackUserAsync(account.UserId);
+            await _accountRepository.DeleteAsync(account);
+            throw;
+        }
+        
+        var mail = new MailForStuffConfirmationRequest()
+        {
+            ToEmail = request.Email,
+            FirstName = request.FirstName,
+            LastName = request.LastName,
+            MiddleName = request.MiddleName,
+            AccountId = account.Id,
+            Password = password
+        };
+
+        await _mailService.SendEmailAsync(mail);
+        return _mapper.Map<GetDetailedReceptionistProfilesResponse>(receptionist);
+    }
+
+    public async Task DeleteAsync(Guid id)
+    {
+        var receptionist = await _receptionistRepository.GetByIdAsync(id);
+        if (receptionist == null)
+            throw new BadHttpRequestException("Receptionist not found");
+
+        await _receptionistRepository.DeleteAsync(receptionist);
+    }
+    
+    public async Task<GetDetailedReceptionistProfilesResponse> GetByUserIdAsync(Guid userId)
+    {
+        var accountReceptionist = await _accountRepository.FindByCondition(x=>x.UserId==userId,trackChanges:false).Include(x=>x.Receptionist).SingleOrDefaultAsync();
+        if (accountReceptionist == null || accountReceptionist.Receptionist == null)
+            throw new BadHttpRequestException("Receptionist not found");
+
+        var receptionist = await _receptionistRepository.GetByIdAsync(accountReceptionist.Receptionist.Id);
+        if(receptionist == null)
+            throw new BadHttpRequestException("Receptionist not found");
+        
+        return _mapper.Map<GetDetailedReceptionistProfilesResponse>(receptionist);
+    }
+    
+    public async Task<GetDetailedReceptionistProfilesResponse> GetByIdAsync(Guid id)
+    {
+        var receptionist = await _receptionistRepository.GetByIdAsync(id);
+        if(receptionist == null)
+            throw new BadHttpRequestException("Receptionist not found");
+        
+        return _mapper.Map<GetDetailedReceptionistProfilesResponse>(receptionist);
+    }
+
+    public async Task<PageResult<GetReceptionistAndPhotoProfilesResponse>> GetAllAsync(int pageNumber, int pageSize)
+    {
```

Comment:
1. It's not ALL. It's GetPage
2. Implement one more mothod on repo to make only one querry to database